plugins/woocommerce/src/Internal/Admin/Settings/PaymentProviders/WooPayments/WooPaymentsService.php

```diff
@@ -668,6 +668,9 @@ public function reset_onboarding( string $from = '', string $source = '' ): arra
 			throw new Exception( esc_html__( 'Failed to reset onboarding.', 'woocommerce' ) );
 		}
 
+		// Remove NOX-specific onboarding data.
+		$this->proxy->call_function( 'delete_option', self::NOX_PROFILE_OPTION_KEY );
```

Comment:
@vladolaru Given you have more knowledge in this area, let me know if this is enough or if something else should be cleaned :)

Reply:
@elazzabi that should do it. It is the nuclear option, but there is no need to be more granular at this point.